mage_ai/server/app.py

```diff
@@ -314,16 +319,19 @@ def clean_df(df, name):
     return (feature_set, result['df'])
 
 
-def clean_df_with_pipeline(df, id=None, path=None):
+def clean_df_with_pipeline(df, id=None, path=None, remote_id=None, mage_api_key=None):
     pipeline = None
     if id is not None:
-        pipeline = Pipeline(id=id)
+        pipeline = Pipeline(id=id).pipeline
     elif path is not None:
-        pipeline = Pipeline(path=path)
+        pipeline = Pipeline(path=path).pipeline
+    elif remote_id is not None:
+        final_key = mage_api_key if mage_api_key is not None else api_key
```

Comment:
api_key could be undefined.

Reply:
@wangxiaoyou1993 good call. I updated to set the `api_key` global variable.